lib/Browser.js

```diff
@@ -109,6 +112,21 @@ Browser.prototype._addService = function (service) {
   this.emit('up', service)
 }
 
+Browser.prototype._updateService = function (service) {
+  let cachedService, index
+  this.services.some(function (s, i) {
+    if (dnsEqual(s.fqdn, service.fqdn)) {
+      cachedService = s
+      index = i
+      return true
+    }
+    return false
+  })
+  if (!cachedService) return
+  this.services[index] = service
+  this.emit('update', service)
+}
+
```

Comment:
Just to clarify my understanding of the change. The bonjour library already supported updating the records (e.g., I you would retrieve the txt record after some time after it updated you would receive the updated value), right? E.g., just looking at the test case, this functionality seems supported, https://github.com/homebridge/bonjour/blob/9e3d67bde06af191fd67f123aa4aeade6c61b8c8/test/bonjour.js#L121-L148?

Would it make sense to add a similar unit test (or extend the given test) to verify that the update event is correctly emitted?

Reply:
Well spent the last hour looking into why that test case was successful, while my real world test showed it not working.  And found that the current test implementation exposes an unexpected behaviour of `Service`, and it fires a `Service.up` event on `updateTxt`, so when `service.updateTxt `fires,  the whole test block is recreated in memory, with a second browser instance, and its the second browser instance that is currently allowing the test to succeed.

Going to redo that particular test case to match real world behaviour, and add a second test case for the update event.  Will also update the README to mention the behaviour of the service.up event.

Reply:
Great catch. Thanks for doing all this analysis 🚀 
The test cases are just copied from the original fork. Rewriting the tests and ensuring proper functionality sounds like a good plan 👍

Thank you for all this hard work!

Reply:
It was an interesting little investigation, and well worth the time spent.